tests/models/test_token_classification_with_seq2seq_encoder_and_crf.py

```diff
@@ -0,0 +1,258 @@
+    seq_len,
+    add_dummy_linear=False,
+    **model_kwargs,
+):
+    class MockConfig:
+        def __init__(
+            self,
+            hidden_size: int = 10,
+            classifier_dropout: float = 0.1,
+            model_type=model_type,
+        ) -> None:
+            self.hidden_size = hidden_size
+            self.model_type = model_type
+            if self.model_type == "bert":
+                self.hidden_dropout_prob = classifier_dropout
+            elif self.model_type == "albert":
+                self.classifier_dropout_prob = classifier_dropout
+            elif self.model_type == "distilbert":
+                self.seq_classif_dropout = classifier_dropout
+            elif self.model_type == "longformer":
+                self.hidden_dropout_prob = classifier_dropout
+            else:
+                self.classifier_dropout = classifier_dropout
+
+    class MockModel(torch.nn.Module):
+        def __init__(self, batch_size, seq_len, hidden_size, add_dummy_linear) -> None:
+            super().__init__()
+            self.batch_size = batch_size
+            self.seq_len = seq_len
+            self.hidden_size = hidden_size
+            if add_dummy_linear:
+                self.dummy_linear = torch.nn.Linear(self.hidden_size, 99)
+
+        def __call__(self, *args, **kwargs):
+            last_hidden_state = torch.FloatTensor(
+                torch.rand(self.batch_size, self.seq_len, self.hidden_size)
+            )
+            return BaseModelOutputWithPoolingAndCrossAttentions(
+                last_hidden_state=last_hidden_state,
+            )
+
+    hidden_size = 10
+
+    monkeypatch.setattr(
+        transformers.AutoConfig,
+        "from_pretrained",
+        lambda model_name_or_path: MockConfig(
+            hidden_size=hidden_size, classifier_dropout=0.1, model_type=model_type
+        ),
+    )
+    monkeypatch.setattr(
+        transformers.AutoModel,
+        "from_pretrained",
+        lambda model_name_or_path, config: MockModel(
+            batch_size=batch_size,
+            seq_len=seq_len,
+            hidden_size=hidden_size,
+            add_dummy_linear=add_dummy_linear,
+        ),
+    )
+
+    # set seed to make the classifier deterministic
+    torch.manual_seed(42)
+    result = TokenClassificationModelWithSeq2SeqEncoderAndCrf(
+        model_name_or_path="bert",
+        num_classes=num_classes,
+        warmup_proportion=0.0,
+        **model_kwargs,
+    )
+    assert not result.is_from_pretrained
+
+    return result
+
+
+@pytest.fixture
+def model(monkeypatch, batch):
+    inputs, targets = batch
+    model = get_model(
+        monkeypatch=monkeypatch,
+        model_type="bert",
+        batch_size=inputs["input_ids"].shape[0],
+        seq_len=inputs["input_ids"].shape[1],
+        num_classes=int(torch.max(targets) + 1),
+    )
+    # model = TokenClassificationModelWithSeq2SeqEncoderAndCrf(
+    #     model_name_or_path="bert-base-uncased",
+    #     num_classes=5,
+    # )
```

Comment:
what's this? If it is still useful, add a comment describing that, otherwise remove it.